File: unit-3/sprint-2/evaluation/Fixeme/app/src/main/java/com/masai/fixme_1/MainActivity.java
package com.masai.fixme_1;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.progressindicator.CircularProgressIndicator;

public class MainActivity extends AppCompatActivity {

    private Button mBtnStartTask;
    private CircularProgressIndicator circularProgressIndicator;

    private Handler mainThreadHandler = new Handler(Looper.getMainLooper()) {
        @Override
        public void handleMessage(@NonNull Message msg) {
            super.handleMessage(msg);
            int progress = (int) msg.obj;
            circularProgressIndicator.setProgress(progress);
            if (progress== 100) {
                Toast.makeText(MainActivity.this,"Your task is completed",Toast.LENGTH_SHORT).show();
                circularProgressIndicator.setVisibility(View.GONE);
            }
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();

    }

    private void initViews() {
        mBtnStartTask= findViewById(R.id.btnStartTask);
        circularProgressIndicator = findViewById(R.id.progress_circular);
        mBtnStartTask.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this,"Your Task Started please wait",Toast.LENGTH_SHORT).show();
                Task task = new Task("async", mainThreadHandler);
                task.start();
            }
        });
    }
}

File: unit-3/sprint-4/day-2/you/18Aug_Animals_profile_in_Fragment_kotlin/app/src/main/java/com/example/a18aug_animals_profile_in_fragment_kotlin/MainActivity.kt
package com.example.a18aug_animals_profile_in_fragment_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.fragment.app.FragmentManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    //private lateinit var fragmentManager: FragmentManager
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnLoad.setOnClickListener{
            whenButtonClicked()
            btnLoad.visibility = View.GONE
        }

    }

    private fun whenButtonClicked() {
        var fragmentManager = supportFragmentManager
        val animalFragment = AnimalFragment()
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.flcontainer,animalFragment).commit()
    }


}